libgadget/utils/paramset.c

```diff
@@ -351,13 +363,14 @@ param_get_string(ParameterSet * ps, char * name)
     return ps->value[p->index].s;
 }
 void
-param_get_string2(ParameterSet * ps, char * name, char * dst)
+param_get_string2(ParameterSet * ps, char * name, char * dst, size_t len)
 {
     ParameterSchema * p = param_get_schema(ps, name);
     if (param_is_nil(ps, name)) {
         printf("Accessing an undefined parameter `%s`.\n", p->name);
     }
-    strcpy(dst, ps->value[p->index].s);
+    strncpy(dst, ps->value[p->index].s, len);
```

Comment:
I think we shall die instead of truncation when the receiver is not large enough.